config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml

```diff
@@ -3,12 +3,14 @@ kind: OpenStackDataPlaneService
 metadata:
   name: nova
 spec:
-  secrets:
-    - nova-cell1-compute-config
-    # NOTE: this Secret needs to be created before deploying the data plane.
-    # It should contain an ssh key-pair in the secret fields: ssh-privatekey
-    # and ssh-publickey
-    - nova-migration-ssh-key
+  dataSources:
+    - secretRef:
+        name: nova-cell1-compute-config
+    - secretRef:
+        name: nova-migration-ssh-key
```

Comment:
so this does not seam to be working correctly

https://review.rdoproject.org/zuul/build/4a1dbeba05de46f6ae7fca76d45374b7/log/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/nova-custom-edpm-deployment-openstack-edpm-ipam-hg5vp/logs/openstackansibleee.log#606-608

[compute-0]: FAILED! => {"changed": false, "msg": "Source /var/lib/openstack/config/nova/ssh-publickey not found"}[0m

it is indeed not copied to 

https://logserver.rdoproject.org/20/920/b8451ed34d57ceee4b573ad2e68f77c6ab901b0b/github-check/podified-multinode-edpm-deployment-crc/4a1dbeb/controller/ci-framework-data/logs/38.129.56.99/openstack/config/nova/

so there appreas to be a bug

the default nova dataplane service does not exist and has been replaced with nova-custom by install_yamls

https://logserver.rdoproject.org/20/920/b8451ed34d57ceee4b573ad2e68f77c6ab901b0b/github-check/podified-multinode-edpm-deployment-crc/4a1dbeb/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/crs/openstackdataplaneservices.dataplane.openstack.org/

and install yamls or the ci framework is replacing all the datasouce section

https://logserver.rdoproject.org/20/920/b8451ed34d57ceee4b573ad2e68f77c6ab901b0b/github-check/podified-multinode-edpm-deployment-crc/4a1dbeb/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/crs/openstackdataplaneservices.dataplane.openstack.org/nova-custom.yaml 

which is why this is not working.

so we need to fix that for this to work

Reply:
yep definetly a bug in install_yamls
```
- patch: |-
    - op: replace
      path: /metadata/name
      value: nova-custom
    - op: add
      path: /spec/dataSources
      value:
        - configMapRef:
            name: nova-extra-config
```
https://logserver.rdoproject.org/20/920/b8451ed34d57ceee4b573ad2e68f77c6ab901b0b/github-check/podified-multinode-edpm-deployment-crc/4a1dbeb/controller/ci-framework-data/artifacts/manifests/openstack/dataplane/cr/kustomization.yaml

which results in
```
---
apiVersion: dataplane.openstack.org/v1beta1
kind: OpenStackDataPlaneService
metadata:
  labels:
    created-by: install_yamls
  name: nova-custom
  namespace: openstack
spec:
  caCerts: combined-ca-bundle
  containerImageFields:
  - NovaComputeImage
  - EdpmIscsidImage
  dataSources:
  - configMapRef:
      name: nova-extra-config
  playbook: osp.edpm.nova
```

that is odd because i intentionally sequenced these changes so that the install yamls patch would be after this to avoid issues like that  in ci

@slagle any idea why this is being updated that way?

Reply:
oh you submitted https://github.com/openstack-k8s-operators/install_yamls/pull/846
ya that is not correct 

i was going to do that in https://github.com/openstack-k8s-operators/install_yamls/pull/845/files

should we revert https://github.com/openstack-k8s-operators/install_yamls/pull/846 or will i just rebase my change and try and fix it that way?

i can flip the depends on

Reply:
https://github.com/openstack-k8s-operators/install_yamls/pull/851 shoudl alow use to move forward